database/schema.go

```diff
@@ -0,0 +1,26 @@
+package database
+
+import (
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// schema defines a mapping between a collection name and the indexes that
+// must exist for that collection.
+var schema = map[string][]mongo.IndexModel{
+	collSkylinks: {
+		{
+			Keys:    bson.D{{"skylink", 1}},
+			Options: options.Index().SetName("skylink").SetUnique(true),
+		},
+		{
+			Keys:    bson.D{{"lock_expires", 1}},
```

Comment:
what's this for?

Reply:
This service needs to do two things:
1. Track which server is pinning a given skylink.
2. Ensure that each skylink on record (that needs to be kept alive) is being pinned by a minimum number of servers.

To the second point, if a skylink is underpinned (in the sense that it's pinned by fewer than the minimum number of servers) it will be pinned by Pinner on some server. Since pinning takes some time, we want to indicate that the skylink is being currently pinned by a server, so other servers won't try to pin it as well. This is the purpose of the `lock` field. The `lock` field isn't implemented yet but it made sense for the `lock_expires` field to be added now, so we could use it when writing all mongo queries that we need today.